solutions/USACO open16-diamond-bronze.mdx

```diff
@@ -0,0 +1,59 @@
+---
+id: usaco-639
+source: USACO Bronze 2016 Open
+title: Diamond Collector
+author: Danh Ta Chi Thanh
+---
+
+[Official Analysis (Java)](http://www.usaco.org/current/data/sol_diamond_bronze_open16.html)
+
+
+
+<LanguageSection>
+
+<CPPSection>
+
+## C++ Implementation
+
+```cpp
+#include <iostream>
+#include <cstdio>
+#include <algorithm>
+#include <vector>
+
+using namespace std;
+
+int n,k,ans,cnt; vector<int> a;
+
+void solve(){
+	cin >> n >> k;
+	a.resize(n);
+	for (int i=0;i<n;++i){
+		cin >> a[i];
+	}
+	sort(a.begin(),a.end());
+	for (int i=0;i<n;++i){
+		cnt=1;
+		for (int j=i+1;j<n;++j){
+			if (a[j]-a[i]<=k) ++cnt;
+			else break;
+		}
+		ans=max(cnt,ans);
+	}
+	cout << ans;
+	return;
+}
+
+int main(){
+	cin.tie(0)->sync_with_stdio(0);
+	freopen("diamond.in","r",stdin);
+	freopen("diamond.out","w",stdout);
+	solve();
+	return 0;
+}
+
+```
+
+</CPPSection>
+
+</LanguageSection>

```

Comment:
All the solutions look good! Since you there aren't multiple tests in one input, I would prefer that you delete the solve function and put your code directly inside the main function.

Reply:
Also if you can add some comments to your code it would be great.

Reply:
I think this is quite unnecessary since the editorial explains the general idea thoroughly, so I reckon that the users should consult the official analysis before reading my code implementation

Reply:
Should I add the "Time complexity" line for each solution?

Reply:
You can if you want, but is not necessary.